include/alp/reference/io.hpp

```diff
@@ -45,6 +45,95 @@
 
 namespace alp {
 
+	/**
+	 * Request the size (dimension) of a given Vector.
+	 *
+	 * The dimension is set at construction of the given Vector and cannot
+	 * be changed. A call to this function shall always succeed.
+	 *
+	 * @tparam DataType      The type of elements contained in the vector \a x.
+	 * @tparam DataStructure The structure of the vector \a x.
+	 * @tparam View          The view type applied to the vector \a x.
+	 *
+	 * @param[in] x The Vector of which to retrieve the size.
+	 *
+	 * @return The size of the Vector \a x.
+	 *
+	//  * \parblock
+	//  * \par Performance semantics
+	//  * A call to this function
+	//  *  -# consists of \f$ \Theta(1) \f$ work;
+	//  *  -# moves \f$ \Theta(1) \f$ bytes of memory;
+	//  *  -# does not allocate any dynamic memory;
+	//  *  -# shall not make any system calls.
+	//  * \endparblock
+	 */
+	template<
+		typename DataType, typename DataStructure, typename View, typename ImfR, typename ImfC
+	>
+	size_t size(
+		const Vector< DataType, DataStructure, Density::Dense, View, ImfR, ImfC, reference > &x
+	) noexcept {
+		return getLength( x );
+	}
+
+	/**
+	 * Request the number of nonzeroes in a given Vector.
+	 *
+	 * A call to this function always succeeds.
+	 *
+	 * @tparam DataType      The type of elements contained in the vector \a x.
+	 * @tparam DataStructure The structure of the vector \a x.
+	 * @tparam View          The view type applied to the vector \a x.
+	 *
+	 * @param[in] x The Vector of which to retrieve the number of nonzeroes.
+	 *
+	 * @return The number of nonzeroes in \a x.
+	 *
+	//  * \parblock
+	//  * \par Performance semantics
+	//  * A call to this function
+	//  *   -# consists of \f$ \Theta(1) \f$ work;
+	//  *   -# moves \f$ \Theta(1) \f$ bytes of memory;
+	//  *   -# does not allocate nor free any dynamic memory;
+	//  *   -# shall not make any system calls.
+	//  * \endparblock
+	 */
+	template<
+		typename DataType, typename DataStructure, typename View, typename ImfR, typename ImfC
+	>
+	size_t nnz(
+		const Vector< DataType, DataStructure, Density::Dense, View, ImfR, ImfC, reference > &x
+	) noexcept {
+		throw std::runtime_error( "Needs an implementation." );
```

Comment:
Should we drop this and have the base return an RC?

Reply:
Agree, we should drop these from reference. But the base versions cannot return an RC because the return type is `size_t` by the spec. In general, such operations assert and return `SIZE_MAX`.